x/genutil/client/cli/migrate.go

```diff
@@ -137,3 +71,75 @@ $ %s migrate v0.36 /path/to/genesis.json --chain-id=cosmoshub-3 --genesis-time=2
 
 	return cmd
 }
+
+// MigrateHandler handles the migration command with a migration map as input,
+// returning an error upon failure.
+func MigrateHandler(cmd *cobra.Command, args []string, migrations types.MigrationMap) error {
+	clientCtx := client.GetClientContextFromCmd(cmd)
+
+	var err error
+
+	target := args[0]
+	importGenesis := args[1]
+
+	genDoc, err := validateGenDoc(importGenesis)
+	if err != nil {
+		return err
+	}
+
+	// Since some default values are valid values, we just print to
+	// make sure the user didn't forget to update these values.
+	if genDoc.ConsensusParams.Evidence.MaxBytes == 0 {
+		fmt.Printf("Warning: consensus_params.evidence.max_bytes is set to 0. If this is"+
+			" deliberate, feel free to ignore this warning. If not, please have a look at the chain"+
+			" upgrade guide at %s.\n", chainUpgradeGuide)
+	}
+
+	var initialState types.AppMap
+	if err := json.Unmarshal(genDoc.AppState, &initialState); err != nil {
+		return errors.Wrap(err, "failed to JSON unmarshal initial genesis state")
+	}
+
+	migrationFunc := GetMigrationCallback(target)
```

Comment:
@julienrbrt seems we need to use [migrations](https://github.com/cosmos/cosmos-sdk/pull/17302/commits/eddff3c4134be5a9d70d22208b4cc7f302ea7925) or pass as [cb](https://github.com/cosmos/cosmos-sdk/pull/17302/commits/5d510c3db3acbbaf84814652739fdc26f8c77ed0)

Reply:
Right!